docs/source/topics/data_types.rst

```diff
@@ -492,8 +495,92 @@ Example of creating an ``PortableCode``:
+* ``engine_command``: The bash command to invoke the container image (e.g. ``singularity exec --bind $PWD:$PWD {image_name}``).
+  The exact form of this command will depend on the containerization technology that is used.
+* ``computer``: The :class:`~aiida.orm.computers.Computer` on which to run the container.
+
+.. important::
+
+    If the ``engine_command`` contains variables (such as in the ``singularity exec --bind $PWD:$PWD {image_name}`` example), it is crucial that the ``Computer`` needs to have the ``use_double_quotes`` setting set to ``True``.
+    By default, a ``Computer`` will use single quotes to escape command line arguments and so the ``$PWD`` would not be expanded.
+    If this wasn't defined as such when the ``Computer`` was created, it can be changed through the API:
+
+    .. code-block::
+
+        computer = load_computer('some-computer')
+        computer.set_use_double_quotes(True)
+
+.. note::
+
+    The container image can be already download and stored on the target computer or can reference an image on a container index, such as Docker Hub.
+    In the latter case, the first time that the containerized code is run, the image will be automatically pulled from the index, which may take a while if the image is large.
+
+
+.. _topics:data_types:core:code:installed:containerized:setup:
+
+Setup
+^^^^^
+
+A ``ContainerizedCode`` can be created through the CLI as well as the API.
+The following examples show how to setup running ``bash`` in a base Docker container through Singularity to be run on the ``Computer`` named ``some-computer``:
+
+.. tab-set::
+
+    .. tab-item:: CLI
+
+        .. code-block:: console
+
+            verdi code create core.code.containerized \
+                --non-interactive \
+                --label containerized-code \
+                --default-calc-job-plugin core.arithmetic.add \
+                --computer some-computer \
+                --filepath-executable "/bin/sh" \
+                --image-name "docker://alpine:3" \
+                --engine-command "singularity exec --bind $PWD:$PWD {image_name}"
+
+    .. tab-item:: API
+
+        .. code-block:: python
+
+            from aiida.orm import ContainerizedCode, load_computer
+
+            code = ContainerizedCode(
+                computer=load_computer('some-computer')
+                filepath_executable='/bin/sh'
+                image_name='docker://alpine:3',
+                engine_command='singularity exec --bind $PWD:$PWD {image_name}'
+            ).store()
+
+
+.. _topics:data_types:core:code:installed:containerized:support:
+
+Supported container technologies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``ContainerizedCode`` is compatible with a variety of containerization technologies:
+
+* `Singularity <https://singularity-docs.readthedocs.io/en/latest/>`__
+* `Sarus <https://sarus.readthedocs.io/en/stable/>`__
+* `NVIDIA/enroot <https://github.com/NVIDIA/enroot/>`__
```

Comment:
on second thought, vanilla `enroot` may fall into the `docker` category after all (not supported), since you would need to call `mpirun` _inside_  the container...

are there any plans to support such scenarios?

P.S. It actually already possible to use enroot containers with AiiDA via the pyxis plugin for slurm, which adds some extra slurm pragmas like
```
#SBATCH --container-image ...
#SBATCH --container-mounts
```
One can then tell these to AiiDa as extra scheduler parameters.
It's not great from a provenance perspective, though, since the mounts, image, etc. are all recorded in a slurm-specific way.

Reply:
Sure

Reply:
I removed the mention of enroot for now until we figure out if and how it can be supported.